tests/h/search/index_test.py

```diff
@@ -6,22 +6,21 @@
 import elasticsearch1_dsl
 import pytest
 
+from h.search import persistence
 from tests.common.matchers import Matcher
 
 
 @pytest.mark.usefixtures("annotations")
 class TestIndex(object):
     def test_annotation_ids_are_used_as_elasticsearch_ids(self, es_client,
                                                           factories,
-                                                          index):
+                                                          index_new):
         annotation = factories.Annotation.build()
 
-        index(annotation)
+        index_new(annotation)
 
-        result = es_client.conn.get(index=es_client.index,
-                                    doc_type="annotation",
-                                    id=annotation.id)
-        assert result["_id"] == annotation.id
+        result = persistence.Annotation.get(id=annotation.id)
+        assert result.meta.id == annotation.id
```

Comment:
Here I've just changed this test to work against Es 6, but what we really want to do is run the test against both Es's.

The API of the `index_new()` fixture is the same as the old `index()` one, so you could parametrize the test with both fixtures.

But getting an annotation back out of Es 6 is different, and the kind of object that you get out is different too.

Reply:
The tests ultimately need an API to work with that is the same for both ES versions. Given that we only need to support a small number of operations (index annotations, get stored doc by ID, return IDs of documents that match a search query) I think it should be reasonably straightforward to provide fixtures that abstract over this.

This is very rough, but something like this perhaps? -

```diff
diff --git a/tests/h/search/index_test.py b/tests/h/search/index_test.py
index 38811a951..cc7d6a0eb 100644
--- a/tests/h/search/index_test.py
+++ b/tests/h/search/index_test.py
@@ -3,24 +3,27 @@ from __future__ import unicode_literals
 
 import datetime
 
+import elasticsearch_dsl
 import elasticsearch1_dsl
 import pytest
 
-from h.search import persistence
 from tests.common.matchers import Matcher
 
 
 @pytest.mark.usefixtures("annotations")
 class TestIndex(object):
+
+    @pytest.mark.parametrize('use_old_es', (True, False))
     def test_annotation_ids_are_used_as_elasticsearch_ids(self, es_client,
                                                           factories,
-                                                          index_new):
+                                                          index,
+                                                          get,
+                                                          use_old_es):
         annotation = factories.Annotation.build()
 
-        index_new(annotation)
+        index(annotation, use_old_es=use_old_es)
 
-        result = persistence.Annotation.get(id=annotation.id)
-        assert result.meta.id == annotation.id
+        assert get(annotation.id, use_old_es=use_old_es) is not None
 
     def test_it_can_index_an_annotation_with_no_document(self, factories,
                                                          index, get):
@@ -111,13 +114,30 @@ class TestIndex(object):
 
     @pytest.fixture
     def get(self, es_client):
-        def _get(annotation_id):
+        def _get(annotation_id, use_old_es=True):
+            if use_old_es:
+                conn = es_client.conn
+                index = es_client.index
+                doc_type = "annotation"
+            else:
+                conn = elasticsearch_dsl.connections.get_connection()
+                index = 'hypothesis-test'
+                doc_type = "doc"
+
             """Return the annotation with the given ID from Elasticsearch."""
-            return es_client.conn.get(
-                index=es_client.index, doc_type="annotation",
-                id=annotation_id)["_source"]
+            return conn.get(index, doc_type=doc_type, id=annotation_id)["_source"]
         return _get
 
+    @pytest.fixture
+    def index(self, index, index_new):
+        def _index(*args, **kwargs):
+            use_old_es = kwargs.get('use_old_es', True)
+            if use_old_es:
+                index(*args)
+            else:
+                index_new(*args)
+        return _index
+
     @pytest.fixture
     def search(self, es_client):
         return elasticsearch1_dsl.Search(using=es_client.conn).fields([])
```

Reply:
To Rob's point I think this would be a good use case for parametrize where you parametrize the old and the new get and index. Rather than having an if else statement like Rob's code above I would recommend just having two fixtures get_old and get_new, index_old, index_new and parametrize those both on the test so:
```
@pytest.mark.parametrize('index,get', ((index_new, get_new), (index_old, get_old)))
```